plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -697,6 +675,20 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                 .iterator();
     }
 
+    private List<ColumnMetadata> getColumnsMetadata(MetadataEntry metadataEntry)
+    {
+        DeltaTable table = DeltaTable.of(metadataEntry);
+        ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
+        for (DeltaLakeColumnHandle column : getColumns(metadataEntry)) {
+            String name = column.getBaseColumnName();
+            switch (column.getColumnType()) {
+                case REGULAR, PARTITION_KEY -> columns.add(getColumnMetadata(column, table.getColumnComment(name), table.isNullable(name), table.getColumnGeneration(name)));
+                case SYNTHESIZED -> columns.add(ColumnMetadata.builder().setName(name).setType(column.getType()).setHidden(true).build());
+            }
```

Comment:
Can we use https://github.com/trinodb/trino/blob/30e116abaeb740812ce387fec696a06485959ad5/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L3212 here?

Reply:
The reason I avoided calling `getColumnMetadata` for synthesized columns  is same as https://github.com/trinodb/trino/pull/18133#discussion_r1266586806